File: src/PointLight.hs
module PointLight (newPointLight, getIlluminationAt) where

data PointLight = { position  :: Vec3
                  , color     :: Color
                  , intensity :: Float
                  }

newPointLight :: Vec3 -> Color -> Float -> PointLight
newPointLight pos color intensity = PointLight pos color $ max 0.0 intensity

getIlluminationAt :: PointLight -> Float -> Color
getIlluminationAt l dist = Vec3.scale (color l) (intensity l / (4 * pi * dist * dist))

File: src/java/PointLight.java
public class PointLight {
	private Vector3 position;
	private ColorRGB color;
	private double intensity;

    public PointLight(Vector3 position, ColorRGB colour, double intensity) {
		this.position = position;
		this.colour = colour;
		this.intensity = intensity;
	}

	public Vector3 getPosition() {
		return position;
	}

	public ColorRGB getColour() {
		return colour;
	}

	public double getIntensity() {
		return intensity;
	}

    public ColorRGB getIlluminationAt(double distance) {
        return color.scale(intensity / (4 * Math.PI * distance * distance));
    }
}
